pkg/controller/job/job_controller.go

```diff
@@ -1304,6 +1316,7 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 		}
 		if jobFinished {
 			jm.recordJobFinished(jobCtx.job, jobCtx.finishedCondition)
+			jm.completedJobStore.add(jobCtx.job.UID)
```

Comment:
This adds it regardless if completed successfully or failed. So, is there a similar issue for failed pods? 

What would be the pros & cons for accounting for succeeded & failed vs only succeeded?

If we want to account also for failed jobs then I would suggest to call it `finishedJobStore`.

Reply:
Since the issue is because of race between the events missing and doesn't happen frequently, we won't observe for failed  jobs unless we have a lot of jobs constantly failing. Because of that we haven't observed for the failed jobs but based on the code it should be the same for it also.

Reply:
Would it be feasible to repro the bug for a filing job in the new unit test? Then we could parametrize the test with the status of the Pod, and we would have an evidence this is needed indeed, other than static code analysis.